src/arch/intel.c

```diff
@@ -57,8 +40,9 @@ static void do_cpuid(unsigned int *eax, unsigned int *ebx, unsigned int *ecx,
 int ceph_arch_intel_probe(void)
 {
 	/* i know how to check this on x86_64... */
-	unsigned int eax = 1, ebx, ecx, edx;
-	do_cpuid(&eax, &ebx, &ecx, &edx);
+	unsigned int eax, ebx, ecx=0, edx=0;
```

Comment:
and i don't understand why we need to assign 0 to ecx and edx. but not to eax, ebx. they are always assigned if `__get_cpuid()` returns 1